We'll work with a dataset of used cars from eBay Kleinanzeigen, a classifieds section of the German eBay website. We sampled 50,000 data points from the full dataset

In [2]:
import pandas as pd
import numpy as np

In [3]:
autos = pd.read_csv(r'C:\Users\Alisson\.anaconda\autos.csv', encoding='latin1')
print(autos)

                dateCrawled                                          name  \
0       2016-03-24 11:52:17                                    Golf_3_1.6   
1       2016-03-24 10:58:45                          A5_Sportback_2.7_Tdi   
2       2016-03-14 12:52:21                Jeep_Grand_Cherokee_"Overland"   
3       2016-03-17 16:54:04                            GOLF_4_1_4__3TÜRER   
4       2016-03-31 17:25:20                Skoda_Fabia_1.4_TDI_PD_Classic   
...                     ...                                           ...   
371523  2016-03-14 17:48:27                    Suche_t4___vito_ab_6_sitze   
371524  2016-03-05 19:56:21         Smart_smart_leistungssteigerung_100ps   
371525  2016-03-19 18:57:12            Volkswagen_Multivan_T4_TDI_7DC_UY2   
371526  2016-03-20 19:41:08                        VW_Golf_Kombi_1_9l_TDI   
371527  2016-03-07 19:39:19  BMW_M135i_vollausgestattet_NP_52.720____Euro   

        seller offerType  price   abtest vehicleType  yearOfRegistration  \

In [4]:
autos.info()
autos.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 371528 entries, 0 to 371527
Data columns (total 20 columns):
 #   Column               Non-Null Count   Dtype 
---  ------               --------------   ----- 
 0   dateCrawled          371528 non-null  object
 1   name                 371528 non-null  object
 2   seller               371528 non-null  object
 3   offerType            371528 non-null  object
 4   price                371528 non-null  int64 
 5   abtest               371528 non-null  object
 6   vehicleType          333659 non-null  object
 7   yearOfRegistration   371528 non-null  int64 
 8   gearbox              351319 non-null  object
 9   powerPS              371528 non-null  int64 
 10  model                351044 non-null  object
 11  kilometer            371528 non-null  int64 
 12  monthOfRegistration  371528 non-null  int64 
 13  fuelType             338142 non-null  object
 14  brand                371528 non-null  object
 15  notRepairedDamage    299468 non-nu

dateCrawled                            name  seller offerType  \
0  2016-03-24 11:52:17                      Golf_3_1.6  privat   Angebot   
1  2016-03-24 10:58:45            A5_Sportback_2.7_Tdi  privat   Angebot   
2  2016-03-14 12:52:21  Jeep_Grand_Cherokee_"Overland"  privat   Angebot   
3  2016-03-17 16:54:04              GOLF_4_1_4__3TÜRER  privat   Angebot   
4  2016-03-31 17:25:20  Skoda_Fabia_1.4_TDI_PD_Classic  privat   Angebot   

   price abtest vehicleType  yearOfRegistration    gearbox  powerPS  model  \
0    480   test         NaN                1993    manuell        0   golf   
1  18300   test       coupe                2011    manuell      190    NaN   
2   9800   test         suv                2004  automatik      163  grand   
3   1500   test  kleinwagen                2001    manuell       75   golf   
4   3600   test  kleinwagen                2008    manuell       69  fabia   

   kilometer  monthOfRegistration fuelType       brand notRepairedDamage  \
0     150000                    0   benzin  volkswagen               NaN   
1     125000                    5   diesel        audi                ja   
2     125000                    8   diesel        jeep               NaN   
3     150000                    6   benzin  volkswagen              nein   
4      90000                    7   diesel       skoda              nein   

           dateCreated  nrOfPictures  postalCode             lastSeen  
0  2016-03-24 00:00:00             0       70435  2016-04-07 03:16:57  
1  2016-03-24 00:00:00             0       66954  2016-04-07 01:46:50  
2  2016-03-14 00:00:00             0       90480  2016-04-05 12:47:46  
3  2016-03-17 00:00:00             0       91074  2016-03-17 17:40:17  
4  2016-03-31 00:00:00             0       60437  2016-04-06 10:17:21

This dataset contains 371,528 rows and 20 columns. The data is related to vehicle listings, from a web scraping process, capturing various attributes of cars being sold. There are 7 columns with interger types like prive, yearOfRegistration, kilometers and there are 13 columns with object type like name, seller, vehicle types. Some columns contain missing values especially for vehicle details like vehicle type, fuel type and gear box

In [6]:
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'kilometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [7]:
autos.rename(columns = {"yearOfRegistration": "registration_year",
             "monthOfRegistration": "registration_month",
             "notRepairedDamage": "unrepaired_damage",
             "dateCreated": "ad_created",
                       "dateCrawled": "date_crawled",
                       "offerType":"offer_type",
                       "vehicleType": "vehicle_type",
                       "fuelType":"fuel_type",
                       "nrOfPictures":"nr_of_pictures",
                       "postalCode":"postal_code",
                       "lastSeen":"last_seen"}, inplace=True)

In [8]:
autos.columns

Index(['date_crawled', 'name', 'seller', 'offer_type', 'price', 'abtest',
       'vehicle_type', 'registration_year', 'gearbox', 'powerPS', 'model',
       'kilometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'nr_of_pictures', 'postal_code',
       'last_seen'],
      dtype='object')

In [9]:
autos.head()

date_crawled                            name  seller offer_type  \
0  2016-03-24 11:52:17                      Golf_3_1.6  privat    Angebot   
1  2016-03-24 10:58:45            A5_Sportback_2.7_Tdi  privat    Angebot   
2  2016-03-14 12:52:21  Jeep_Grand_Cherokee_"Overland"  privat    Angebot   
3  2016-03-17 16:54:04              GOLF_4_1_4__3TÜRER  privat    Angebot   
4  2016-03-31 17:25:20  Skoda_Fabia_1.4_TDI_PD_Classic  privat    Angebot   

   price abtest vehicle_type  registration_year    gearbox  powerPS  model  \
0    480   test          NaN               1993    manuell        0   golf   
1  18300   test        coupe               2011    manuell      190    NaN   
2   9800   test          suv               2004  automatik      163  grand   
3   1500   test   kleinwagen               2001    manuell       75   golf   
4   3600   test   kleinwagen               2008    manuell       69  fabia   

   kilometer  registration_month fuel_type       brand unrepaired_damage  \
0     150000                   0    benzin  volkswagen               NaN   
1     125000                   5    diesel        audi                ja   
2     125000                   8    diesel        jeep               NaN   
3     150000                   6    benzin  volkswagen              nein   
4      90000                   7    diesel       skoda              nein   

            ad_created  nr_of_pictures  postal_code            last_seen  
0  2016-03-24 00:00:00               0        70435  2016-04-07 03:16:57  
1  2016-03-24 00:00:00               0        66954  2016-04-07 01:46:50  
2  2016-03-14 00:00:00               0        90480  2016-04-05 12:47:46  
3  2016-03-17 00:00:00               0        91074  2016-03-17 17:40:17  
4  2016-03-31 00:00:00               0        60437  2016-04-06 10:17:21

In the previous code, the columns have been modifide for readability

In [11]:
autos.describe(include="all")

date_crawled         name  seller offer_type         price  \
count                371528       371528  371528     371528  3.715280e+05   
unique               280500       233531       2          2           NaN   
top     2016-03-24 14:49:47  Ford_Fiesta  privat    Angebot           NaN   
freq                      7          657  371525     371516           NaN   
mean                    NaN          NaN     NaN        NaN  1.729514e+04   
std                     NaN          NaN     NaN        NaN  3.587954e+06   
min                     NaN          NaN     NaN        NaN  0.000000e+00   
25%                     NaN          NaN     NaN        NaN  1.150000e+03   
50%                     NaN          NaN     NaN        NaN  2.950000e+03   
75%                     NaN          NaN     NaN        NaN  7.200000e+03   
max                     NaN          NaN     NaN        NaN  2.147484e+09   

        abtest vehicle_type  registration_year  gearbox        powerPS  \
count   371528       333659      371528.000000   351319  371528.000000   
unique       2            8                NaN        2            NaN   
top       test    limousine                NaN  manuell            NaN   
freq    192585        95894                NaN   274214            NaN   
mean       NaN          NaN        2004.577997      NaN     115.549477   
std        NaN          NaN          92.866598      NaN     192.139578   
min        NaN          NaN        1000.000000      NaN       0.000000   
25%        NaN          NaN        1999.000000      NaN      70.000000   
50%        NaN          NaN        2003.000000      NaN     105.000000   
75%        NaN          NaN        2008.000000      NaN     150.000000   
max        NaN          NaN        9999.000000      NaN   20000.000000   

         model      kilometer  registration_month fuel_type       brand  \
count   351044  371528.000000       371528.000000    338142      371528   
unique     251            NaN                 NaN         7          40   
top       golf            NaN                 NaN    benzin  volkswagen   
freq     30070            NaN                 NaN    223857       79640   
mean       NaN  125618.688228            5.734445       NaN         NaN   
std        NaN   40112.337051            3.712412       NaN         NaN   
min        NaN    5000.000000            0.000000       NaN         NaN   
25%        NaN  125000.000000            3.000000       NaN         NaN   
50%        NaN  150000.000000            6.000000       NaN         NaN   
75%        NaN  150000.000000            9.000000       NaN         NaN   
max        NaN  150000.000000           12.000000       NaN         NaN   

       unrepaired_damage           ad_created  nr_of_pictures   postal_code  \
count             299468               371528        371528.0  371528.00000   
unique                 2                  114             NaN           NaN   
top                 nein  2016-04-03 00:00:00             NaN           NaN   
freq              263182                14450             NaN           NaN   
mean                 NaN                  NaN             0.0   50820.66764   
std                  NaN                  NaN             0.0   25799.08247   
min                  NaN                  NaN             0.0    1067.00000   
25%                  NaN                  NaN             0.0   30459.00000   
50%                  NaN                  NaN             0.0   49610.00000   
75%                  NaN                  NaN             0.0   71546.00000   
max                  NaN                  NaN             0.0   99998.00000   

                  last_seen  
count                371528  
unique               182806  
top     2016-04-07 06:45:59  
freq                     17  
mean                    NaN  
std                     NaN  
min                     NaN  
25%                     NaN  
50%                     NaN  
75%                     NaN  
max                  

In [12]:
autos["price"] = autos["price"].astype(int)

In [13]:
autos["price"].describe()

count    3.715280e+05
mean     1.729514e+04
std      3.587954e+06
min      0.000000e+00
25%      1.150000e+03
50%      2.950000e+03
75%      7.200000e+03
max      2.147484e+09
Name: price, dtype: float64

In [14]:
autos.rename(columns = {'odometer':'odometer_km'})

date_crawled                                          name  \
0       2016-03-24 11:52:17                                    Golf_3_1.6   
1       2016-03-24 10:58:45                          A5_Sportback_2.7_Tdi   
2       2016-03-14 12:52:21                Jeep_Grand_Cherokee_"Overland"   
3       2016-03-17 16:54:04                            GOLF_4_1_4__3TÜRER   
4       2016-03-31 17:25:20                Skoda_Fabia_1.4_TDI_PD_Classic   
...                     ...                                           ...   
371523  2016-03-14 17:48:27                    Suche_t4___vito_ab_6_sitze   
371524  2016-03-05 19:56:21         Smart_smart_leistungssteigerung_100ps   
371525  2016-03-19 18:57:12            Volkswagen_Multivan_T4_TDI_7DC_UY2   
371526  2016-03-20 19:41:08                        VW_Golf_Kombi_1_9l_TDI   
371527  2016-03-07 19:39:19  BMW_M135i_vollausgestattet_NP_52.720____Euro   

        seller offer_type  price   abtest vehicle_type  registration_year  \
0       privat    Angebot    480     test          NaN               1993   
1       privat    Angebot  18300     test        coupe               2011   
2       privat    Angebot   9800     test          suv               2004   
3       privat    Angebot   1500     test   kleinwagen               2001   
4       privat    Angebot   3600     test   kleinwagen               2008   
...        ...        ...    ...      ...          ...                ...   
371523  privat    Angebot   2200     test          NaN               2005   
371524  privat    Angebot   1199     test       cabrio               2000   
371525  privat    Angebot   9200     test          bus               1996   
371526  privat    Angebot   3400     test        kombi               2002   
371527  privat    Angebot  28990  control    limousine               2013   

          gearbox  powerPS        model  kilometer  registration_month  \
0         manuell        0         golf     150000                   0   
1         manuell      190          NaN     125000                   5   
2       automatik      163        grand     125000                   8   
3         manuell       75         golf     150000                   6   
4         manuell       69        fabia      90000                   7   
...           ...      ...          ...        ...                 ...   
371523        NaN        0          NaN      20000                   1   
371524  automatik      101       fortwo     125000                   3   
371525    manuell      102  transporter     150000                   3   
371526    manuell      100         golf     150000                   6   
371527    manuell      320      m_reihe      50000                   8   

       fuel_type           brand unrepaired_damage           ad_created  \
0         benzin      volkswagen               NaN  2016-03-24 00:00:00   
1         diesel            audi                ja  2016-03-24 00:00:00   
2         diesel            jeep               NaN  2016-03-14 00:00:00   
3         benzin      volkswagen              nein  2016-03-17 00:00:00   
4         diesel           skoda              nein  2016-03-31 00:00:00   
...          ...             ...               ...                  ...   
371523       NaN  sonstige_autos               NaN  2016-03-14 00:00:00   
371524    benzin           smart              nein  2016-03-05 00:00:00   
371525    diesel      volkswagen              nein  2016-03-19 00:00:00   
371526    diesel      volkswagen               NaN  2016-03-20 00:00:00   
371527    benzin             bmw              nein  2016-03-07 00:00:00   

        nr_of_pictures  postal_code            last_seen  
0                    0        70435  2016-04-07 03:16:57  
1                    0        66954  2016-04-07 01:46:50  
2                    0        90480  2016-04-05 12:47:46  
3                    0        91074  2016-03-17 17:40:17  
4                    0        60437  2016-04-06 10:17:21  
...                ..

In [15]:
autos["price"].unique().shape

(5597,)

In [16]:
autos["price"].describe()

count    3.715280e+05
mean     1.729514e+04
std      3.587954e+06
min      0.000000e+00
25%      1.150000e+03
50%      2.950000e+03
75%      7.200000e+03
max      2.147484e+09
Name: price, dtype: float64

In [17]:
autos["price"].value_counts().head().sort_index(ascending=True)

price
0       10778
500      5670
1000     4649
1200     4594
1500     5394
Name: count, dtype: int64

In [18]:
#"df[df["col"].between(x,y)]
autos[autos["price"].between(500,10985)]

date_crawled  \
2       2016-03-14 12:52:21   
3       2016-03-17 16:54:04   
4       2016-03-31 17:25:20   
5       2016-04-04 17:36:23   
6       2016-04-01 20:48:51   
...                     ...   
371521  2016-03-27 20:36:20   
371523  2016-03-14 17:48:27   
371524  2016-03-05 19:56:21   
371525  2016-03-19 18:57:12   
371526  2016-03-20 19:41:08   

                                                     name  seller offer_type  \
2                          Jeep_Grand_Cherokee_"Overland"  privat    Angebot   
3                                      GOLF_4_1_4__3TÜRER  privat    Angebot   
4                          Skoda_Fabia_1.4_TDI_PD_Classic  privat    Angebot   
5       BMW_316i___e36_Limousine___Bastlerfahrzeug__Ex...  privat    Angebot   
6                             Peugeot_206_CC_110_Platinum  privat    Angebot   
...                                                   ...     ...        ...   
371521                 Opel_Zafira_1.6_Elegance_TÜV_12/16  privat    Angebot   
371523                         Suche_t4___vito_ab_6_sitze  privat    Angebot   
371524              Smart_smart_leistungssteigerung_100ps  privat    Angebot   
371525                 Volkswagen_Multivan_T4_TDI_7DC_UY2  privat    Angebot   
371526                             VW_Golf_Kombi_1_9l_TDI  privat    Angebot   

        price   abtest vehicle_type  registration_year    gearbox  powerPS  \
2        9800     test          suv               2004  automatik      163   
3        1500     test   kleinwagen               2001    manuell       75   
4        3600     test   kleinwagen               2008    manuell       69   
5         650     test    limousine               1995    manuell      102   
6        2200     test       cabrio               2004    manuell      109   
...       ...      ...          ...                ...        ...      ...   
371521   1150  control          bus               2000    manuell        0   
371523   2200     test          NaN               2005        NaN        0   
371524   1199     test       cabrio               2000  automatik      101   
371525   9200     test          bus               1996    manuell      102   
371526   3400     test        kombi               2002    manuell      100   

              model  kilometer  registration_month fuel_type           brand  \
2             grand     125000                   8    diesel            jeep   
3              golf     150000                   6    benzin      volkswagen   
4             fabia      90000                   7    diesel           skoda   
5               3er     150000                  10    benzin             bmw   
6           2_reihe     150000                   8    benzin         peugeot   
...             ...        ...                 ...       ...             ...   
371521       zafira     150000                   3    benzin            opel   
371523          NaN      20000                   1       NaN  sonstige_autos   
371524       fortwo     125000                   3    benzin           smart   
371525  transporter     150000                   3    diesel      volkswagen   
371526         golf     150000                   6    diesel      volkswagen   

       unrepaired_damage           ad_created  nr_of_pictures  postal_code  \
2                    NaN  2016-03-14 00:00:00               0        90480   
3                   nein  2016-03-17 00:00:00               0        91074   
4                   nein  2016-03-31 00:00:00               0        60437   
5                     ja  2016-04-04 00:00:00               0        33775   
6                   nein  2016-04-01 00:00:00               0        67112   
...                  ...                  ...             ...          ...   
371521              nein  2016-03-27 00:00:00               0        26624   
371523               NaN  2016-03-14 00:00:00               0        39576   
371524              nein  2016-03-05 00:00:00               0        26135   
371

Since the dataset contains dataframe of used cars, cars cannot be sold for a price of 0 and the price 0f 349,000 is an outlier.

In [20]:
autos["date_crawled"].str[:10].value_counts(normalize=True, dropna=False).sort_index(ascending=True)

date_crawled
2016-03-05    0.025551
2016-03-06    0.014462
2016-03-07    0.035690
2016-03-08    0.033454
2016-03-09    0.034210
2016-03-10    0.032563
2016-03-11    0.032727
2016-03-12    0.036194
2016-03-13    0.015735
2016-03-14    0.036275
2016-03-15    0.033451
2016-03-16    0.030148
2016-03-17    0.031661
2016-03-18    0.013127
2016-03-19    0.035292
2016-03-20    0.036353
2016-03-21    0.035739
2016-03-22    0.032469
2016-03-23    0.031971
2016-03-24    0.029912
2016-03-25    0.032934
2016-03-26    0.031965
2016-03-27    0.030275
2016-03-28    0.035112
2016-03-29    0.034170
2016-03-30    0.033529
2016-03-31    0.031885
2016-04-01    0.034116
2016-04-02    0.035079
2016-04-03    0.038735
2016-04-04    0.037612
2016-04-05    0.012823
2016-04-06    0.003163
2016-04-07    0.001620
Name: proportion, dtype: float64

In [21]:
autos["ad_created"].str[:10].value_counts(normalize=True, dropna=False).sort_index(ascending=True)

ad_created
2014-03-10    0.000003
2015-03-20    0.000003
2015-06-11    0.000003
2015-06-18    0.000003
2015-08-07    0.000003
                ...   
2016-04-03    0.038893
2016-04-04    0.037741
2016-04-05    0.011655
2016-04-06    0.003155
2016-04-07    0.001558
Name: proportion, Length: 114, dtype: float64

In [22]:
autos["last_seen"].str[:10].value_counts(normalize=True, dropna=False).sort_index(ascending=True)

last_seen
2016-03-05    0.001292
2016-03-06    0.004134
2016-03-07    0.005262
2016-03-08    0.008056
2016-03-09    0.009994
2016-03-10    0.011563
2016-03-11    0.013046
2016-03-12    0.023401
2016-03-13    0.008489
2016-03-14    0.012301
2016-03-15    0.016411
2016-03-16    0.016419
2016-03-17    0.028760
2016-03-18    0.006931
2016-03-19    0.016314
2016-03-20    0.019915
2016-03-21    0.020136
2016-03-22    0.020607
2016-03-23    0.018149
2016-03-24    0.019237
2016-03-25    0.019097
2016-03-26    0.016160
2016-03-27    0.016909
2016-03-28    0.022273
2016-03-29    0.023312
2016-03-30    0.023856
2016-03-31    0.024238
2016-04-01    0.024022
2016-04-02    0.025016
2016-04-03    0.025366
2016-04-04    0.025654
2016-04-05    0.126206
2016-04-06    0.217830
2016-04-07    0.129648
Name: proportion, dtype: float64

In [23]:
autos["registration_year"].describe()

count    371528.000000
mean       2004.577997
std          92.866598
min        1000.000000
25%        1999.000000
50%        2003.000000
75%        2008.000000
max        9999.000000
Name: registration_year, dtype: float64

From the above, maximum year is 9999 and minimum year is 1000, yet earliest years are from 2016 according to data in last seen, ad created and date crawled. Minimum year 2016 and maximum year this year 2024.

In [25]:
autos["registration_year"].value_counts(normalize=True)

registration_year
2000    0.066081
1999    0.061279
2005    0.060065
2006    0.054451
2001    0.054419
          ...   
2066    0.000003
9450    0.000003
3200    0.000003
1925    0.000003
3500    0.000003
Name: proportion, Length: 155, dtype: float64

In [26]:
autos[autos["registration_year"].between(2016,2024)]

date_crawled  \
16      2016-04-01 12:46:46   
22      2016-03-23 14:52:51   
26      2016-03-10 19:38:18   
31      2016-03-29 16:57:02   
35      2016-03-08 07:54:46   
...                     ...   
371349  2016-03-10 23:36:19   
371354  2016-03-17 00:56:26   
371407  2016-03-25 09:37:59   
371444  2016-03-05 14:55:29   
371519  2016-03-09 13:37:43   

                                                     name  seller offer_type  \
16                                            Polo_6n_1_4  privat    Angebot   
22                          Opel_Meriva_1.Hand_TÜV_3.2018  privat    Angebot   
26                              Citroen_C4_Grand_Picasso.  privat    Angebot   
31                           Renault_clio_1.2_TÜV_07/2016  privat    Angebot   
35                                              VW_Golf_3  privat    Angebot   
...                                                   ...     ...        ...   
371349            Mercedes_Benz_CLK_200_Kompressor_6_Gang  privat    Angebot   
371354                       FORD_Fiesta__2.Hand__TÜV_neu  privat    Angebot   
371407                                       Opel_Corsa_C  privat    Angebot   
371444                                     Citroen_jumper  privat    Angebot   
371519  Alfa_Romeo_159_Jtdm_1.9_150_ps_13_600_km_top_voll  privat    Angebot   

        price   abtest vehicle_type  registration_year    gearbox  powerPS  \
16        300     test          NaN               2016        NaN       60   
22       2900     test          NaN               2018    manuell       90   
26       5555  control          NaN               2017    manuell      125   
31        899  control          NaN               2016    manuell       60   
35        350     test          NaN               2016    manuell       75   
...       ...      ...          ...                ...        ...      ...   
371349   1350     test          NaN               2016    manuell        0   
371354   2140     test          NaN               2018    manuell       80   
371407   1250     test          NaN               2018        NaN        0   
371444   5000  control          NaN               2017    manuell      120   
371519   5250  control          NaN               2016  automatik      150   

         model  kilometer  registration_month fuel_type          brand  \
16        polo     150000                   0    benzin     volkswagen   
22      meriva     150000                   5    benzin           opel   
26          c4     125000                   4       NaN        citroen   
31        clio     150000                   6    benzin        renault   
35        golf     150000                   4    benzin     volkswagen   
...        ...        ...                 ...       ...            ...   
371349     clk     150000                   8       NaN  mercedes_benz   
371354  fiesta     150000                   6       NaN           ford   
371407   corsa     150000                   0    benzin           opel   
371444  andere     150000                   7       NaN        citroen   
371519     159     150000                  12       NaN     alfa_romeo   

       unrepaired_damage           ad_created  nr_of_pictures  postal_code  \
16                   NaN  2016-04-01 00:00:00               0        38871   
22                  nein  2016-03-23 00:00:00               0        49716   
26                  nein  2016-03-10 00:00:00               0        31139   
31                   NaN  2016-03-29 00:00:00               0        37075   
35                  nein  2016-03-08 00:00:00               0        19386   
...                  ...                  ...             ...          ...   
371349                ja  2016-03-10 00:00:00               0        26427   
371354              nein  2016-03-17 00:00:00               0        44866   
371407               NaN  2016-03-25 00:00:00               0        45527   
371444                ja  2016-03-05 00:00:00               0        15518 

In [27]:
autos["brand"].index

RangeIndex(start=0, stop=371528, step=1)

In [28]:
brand_dict = {}
for value in autos["brand"]:
    if value not in brand_dict:
        brand_dict[value] = 1
    else:
        brand_dict[value] += 1

print(brand_dict)

{'volkswagen': 79640, 'audi': 32873, 'jeep': 807, 'skoda': 5641, 'bmw': 40274, 'peugeot': 11027, 'ford': 25573, 'mazda': 5695, 'nissan': 5037, 'renault': 17969, 'mercedes_benz': 35309, 'opel': 40136, 'seat': 7022, 'citroen': 5182, 'honda': 2836, 'fiat': 9676, 'mini': 3394, 'smart': 5249, 'hyundai': 3646, 'sonstige_autos': 3982, 'alfa_romeo': 2345, 'subaru': 779, 'volvo': 3327, 'mitsubishi': 3061, 'kia': 2555, 'suzuki': 2328, 'lancia': 484, 'porsche': 2215, 'toyota': 4694, 'chevrolet': 1845, 'dacia': 900, 'daihatsu': 806, 'trabant': 591, 'saab': 530, 'chrysler': 1452, 'jaguar': 621, 'daewoo': 542, 'rover': 490, 'land_rover': 770, 'lada': 225}


In [29]:
autos["brand"].describe()

count         371528
unique            40
top       volkswagen
freq           79640
Name: brand, dtype: object

In [55]:
autos["brand"].unique()

array(['volkswagen', 'audi', 'jeep', 'skoda', 'bmw', 'peugeot', 'ford',
       'mazda', 'nissan', 'renault', 'mercedes_benz', 'opel', 'seat',
       'citroen', 'honda', 'fiat', 'mini', 'smart', 'hyundai',
       'sonstige_autos', 'alfa_romeo', 'subaru', 'volvo', 'mitsubishi',
       'kia', 'suzuki', 'lancia', 'porsche', 'toyota', 'chevrolet',
       'dacia', 'daihatsu', 'trabant', 'saab', 'chrysler', 'jaguar',
       'daewoo', 'rover', 'land_rover', 'lada'], dtype=object)

In [66]:
brand_mean_price = autos.groupby(autos["brand"])["price"].mean().sort_values(ascending=False)
print(brand_mean_price)

brand
sonstige_autos    615647.624309
porsche            51089.857336
alfa_romeo         35737.169723
trabant            20415.935702
mercedes_benz      17244.060438
land_rover         16707.559740
audi               15868.514221
bmw                14844.144883
volkswagen         14533.496007
jaguar             14228.083736
jeep               12009.677819
mini                9846.432233
citroen             8880.624855
volvo               8774.005110
ford                8462.054706
chevrolet           7447.764228
skoda               6411.183478
dacia               5839.654444
mazda               5680.361194
kia                 5656.574168
hyundai             5415.057049
fiat                5326.312836
toyota              5232.657009
nissan              4527.510820
suzuki              4371.683419
seat                4356.666050
subaru              4248.948652
chrysler            3916.025482
saab                3820.041509
honda               3804.630113
smart               3531.562012
mi

From the above results, we can tell sontige autos on average are the most expensive car brand and the least expensive car brand is daewoo

In [69]:
bmp_series = pd.Series(brand_mean_price)
print(bmp_series)

brand
sonstige_autos    615647.624309
porsche            51089.857336
alfa_romeo         35737.169723
trabant            20415.935702
mercedes_benz      17244.060438
land_rover         16707.559740
audi               15868.514221
bmw                14844.144883
volkswagen         14533.496007
jaguar             14228.083736
jeep               12009.677819
mini                9846.432233
citroen             8880.624855
volvo               8774.005110
ford                8462.054706
chevrolet           7447.764228
skoda               6411.183478
dacia               5839.654444
mazda               5680.361194
kia                 5656.574168
hyundai             5415.057049
fiat                5326.312836
toyota              5232.657009
nissan              4527.510820
suzuki              4371.683419
seat                4356.666050
subaru              4248.948652
chrysler            3916.025482
saab                3820.041509
honda               3804.630113
smart               3531.562012
mi

In [79]:
df = pd.DataFrame(brand_mean_price, columns=['mean_price'])
df

Empty DataFrame
Columns: [mean_price]
Index: []

In [81]:
autos.select_dtypes(include='object').columns

Index(['date_crawled', 'name', 'seller', 'offer_type', 'abtest',
       'vehicle_type', 'gearbox', 'model', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'last_seen'],
      dtype='object')

Analysis next steps:
.
Find the most common brand/model combination
.
Split the odometer_km into groups, and use aggregation to see if average prices follows any patterns based on the mileag
..
How much cheaper are cars with damage than their non-damaged counterparts?